### PR TITLE
Browser tests

### LINKED ISSUE #302: Fix behaviour tests
They depend on a Saucelabs account that no longer exists. Instead, we can use headless Chrome. That can run inside Docker and on Travis.

### LINKED ISSUE #339: Headless browser tests for key workflows
Use a tool like https://www.cypress.io/ to create headless browser tests for key workflows like payments, mail, bookings. 

Relates to #307 and indirectly #302. 

## Key workflows to test

registration (profile creation) via:
- [x] new user making a booking request (#370)

room search, selection and booking request submission workflow for:
- [x] new users (just being explicit; this is the same as "new user making a booking request" above) (#370)  
- [x] logged in users (claimed by @bfirsh, in #450 )
- [x] logged out users with accounts (claimed by @bfirsh, in #450 )
 
Payments and booking management
- [x] user adding credit card to their profile (claimed by @bfirsh, in #450  )
- [x] location admin confirms booking and charges card (claimed by @bfirsh, in #450 )
- [x] location admin confirms booking and charges card but payment fails (claimed by @bfirsh, in #450 )
- [ ] location admin can send an email manually
- [x] Refund (claimed by @bfirsh, in #450)

Emails
- [ ] daily resident, guest and admin email functions work when 

## Other workflows

- [ ] new user registering and RSVP-ing for an event

### PR DESCRIPTION
**Pull request by bfirsh**
It's a robot that clicks through the website!

This is using [Cypress](https://www.cypress.io/) which is so so much nicer than the selenium tests of old. [The test results are available on the web! They have screenshots and video to debug!](https://dashboard.cypress.io/#/projects/3gfovh/runs/6/specs) (Link is at end of Travis log output.)

There are various other changes in this PR, so you might want to click through the commits one by one to review. I've organised them carefully for that reason.

fixes #302 
fixes #339

### CODE REVIEW

File: core/views/unsorted.py [hunk 1]
```diff
@@ -2301,15 +2301,17 @@ def process_unsaved_booking(request):
     return
 
 
-def user_login(request):
+def user_login(request, username=None):
```
Comment (jonathan-s) [inline core/views/unsorted.py]: I would suggest that we use https://djangopackages.org/packages/p/django-registration/ for registration / login. If you agree let's open a new issue for that.

Comment (bfirsh) [reply core/views/unsorted.py]: Yes sounds good! These views looked like a bit of a mess but I didn’t want to refactor them now. 🙈

Comment (jonathan-s) [reply core/views/unsorted.py]: There will come a day :)

Comment (jessykate) [reply core/views/unsorted.py]: yes agree! this project was started before django had a registration or profile functionality built in. have been keen to move to the newer functionality.

**Pull request merged**